lib/client/workflow-client/workflowClient.js

```diff
@@ -0,0 +1,367 @@
+  var status;
+  var stepReview = this.stepReview(workflow.steps, result);
+  if (stepReview.nextStepIndex >= workflow.steps.length - 1 && stepReview.complete) {
+    status = CONSTANTS.STATUS.COMPLETE_DISPLAY;
+  } else if (!workorder.assignee) {
+    status = CONSTANTS.STATUS.UNASSIGNED_DISPLAY;
+  } else if (stepReview.nextStepIndex < 0) {
+    status = CONSTANTS.STATUS.NEW_DISPLAY;
+  } else {
+    status = CONSTANTS.STATUS.PENDING_DISPLAY;
+  }
+  return status;
+};
+
+/**
+ *
+ * This function checks each of the result steps to determine if the workflow is complete,
+ * and if not, what is the next step in the workflow to display to the user.
+ *
+ *
+ * @param {object} steps
+ * @param {object} result
+ * @returns {{nextStepIndex: number, complete: *}}
+ */
+WorkflowMediatorService.prototype.stepReview = function stepReview(steps, result) {
+  var nextIncompleteStepIndex = 0;
+  var complete = false;
+
+  //If there is no result, then the first step is the next step.
+  if (result && result.stepResults && result.stepResults.length !== 0) {
+    nextIncompleteStepIndex = _.findIndex(steps, function(step) {
+      //The next incomplete step is the step with no entry or it's not complete yet.
+      return !result.stepResults[step.code] || result.stepResults[step.code].status !== CONSTANTS.STATUS.COMPLETE;
+    });
+
+    if (nextIncompleteStepIndex === -1) {
+      complete = true;
+      nextIncompleteStepIndex = steps.length;
+    }
+  }
+  return {
+    nextStepIndex: nextIncompleteStepIndex,
+    complete: complete // false is any steps are "pending"
+  };
+};
+
+
+/**
+ *
+ * Building a summary of a single workorder.
+ *
+ * In this case, we need access to the workorder, workflow and result objects.
+ *
+ * @param workorderId
+ */
+WorkflowMediatorService.prototype.getWorkorderSummary = function(workorderId) {
+  var self = this;
+
+  return this.readWorkorder(workorderId).then(function(workorder) {
+    return q.all([self.read(workorder.workflowId), self.getResultByWorkorderId(workorderId)]).then(function(allResults) {
+      var workflow = allResults[0];
+      var result = allResults[1];
+
+      return [workorder, workflow, result];
+    });
+  });
+};
+
+/**
+ *
+ * Finding a result based on a workorder ID
+ *
+ * @param {string} workorderId - The ID of the workorder to filter by.
+ */
+WorkflowMediatorService.prototype.getResultByWorkorderId = function getResultByWorkorderId(workorderId) {
+  return this.listResults().then(function(resultsArray) {
+    return _.find(resultsArray || [], function(result) {
+      return result.workorderId === workorderId;
+    }) || null;
+  });
+};
+
+/**
+ *
+ * Creating a new result object for a workorder ID
+ *
+ * @param workorderId
+ */
+WorkflowMediatorService.prototype.createNewResult = function createNewResult(workorderId) {
+  return this.createResult({status: CONSTANTS.STATUS.NEW_DISPLAY, nextStepIndex: 0, workorderId: workorderId, stepResults: {}});
```

Comment:
Please consider breaking this into two lines. We should try to avoid the need to scroll right to read full line.